src/Authentication/Authentication/Common/GraphSessionInitializer.cs

```diff
@@ -30,7 +30,13 @@ internal static GraphSession CreateInstance(IDataStore dataStore = null)
         {
             IGraphOption graphOptions = null;
             // Try to create directory if it doesn't exist.
-            Directory.CreateDirectory(Core.Constants.GraphDirectoryPath);
+            try 
+            {
+                Directory.CreateDirectory(Core.Constants.GraphDirectoryPath);
+            }
+            finally
```

Comment:
I believe you meant to use catch instead of catch. Finally will still bubble to exception to the caller:
```suggestion
            catch
```

Reply:
Ah, right you are. I actually tested this with a catch block originally, but swapped it with finally when I didn't think it was needed. Put it back to catch now.